File: src/rpnumerics/AccumulationFunction.java
/*
 * Instituto de Matematica Pura e Aplicada - IMPA
 * Departamento de Dinamica dos Fluidos
 *
 */

package rpnumerics;

import wave.util.RealVector;
import wave.util.RealMatrix2;
import wave.util.HessianMatrix;

public  interface AccumulationFunction
{
	HessianMatrix D2H( RealVector U );

	RealMatrix2 DH( RealVector U );

	RealVector H( RealVector U );

//	AccumulationParams accumulationParams( );


}

File: src/rpnumerics/AccumulationFunction.cc
#include "AccumulationFunction.h"

inline AccumulationFunction::AccumulationFunction(void) :
	params_(AccumulationParams())
{
}

inline AccumulationFunction::AccumulationFunction(const AccumulationParams & params) :
	params_(params)
{
}

inline AccumulationFunction::~AccumulationFunction(void) 
{
}

inline void AccumulationFunction::accumulationParams(const AccumulationParams & params)
{
	params_ = params;
}

inline const AccumulationParams & AccumulationFunction::accumulationParams(void)
{
	return params_;
}
